In [1]:
import pandas as pd

In [2]:
wine_frame = pd.read_csv('./wine-reviews/wine_data_1.csv')

In [3]:
# Seeing what shape the data is in.
wine_frame[:5]

Unnamed: 0 country                                        description  \
0           0      US  This tremendous 100% varietal wine hails from ...   
1           1   Spain  Ripe aromas of fig, blackberry and cassis are ...   
2           2      US  Mac Watson honors the memory of a wine once ma...   
3           3      US  This spent 20 months in 30% new French oak, an...   
4           4  France  This is the top wine from La Bégude, named aft...   

                            designation  points  price        province  \
0                     Martha's Vineyard      96  235.0      California   
1  Carodorum Selección Especial Reserva      96  110.0  Northern Spain   
2         Special Selected Late Harvest      96   90.0      California   
3                               Reserve      96   65.0          Oregon   
4                            La Brûlade      95   66.0        Provence   

            region_1           region_2             variety  \
0        Napa Valley               Napa  Cabernet Sauvignon   
1               Toro                NaN       Tinta de Toro   
2     Knights Valley             Sonoma     Sauvignon Blanc   
3  Willamette Valley  Willamette Valley          Pinot Noir   
4             Bandol                NaN  Provence red blend   

                    winery  
0                    Heitz  
1  Bodega Carmen Rodríguez  
2                 Macauley  
3                    Ponzi  
4     Domaine de la Bégude

In [4]:
# Create a slice that only Contains Califorina Cabernet Savignon
cali_cab =wine_frame[ wine_frame.province == 'California']
cali_cab =cali_cab[cali_cab.variety =='Cabernet Sauvignon']

In [5]:
#How many Califorina Cabernet are in the Data Set
cali_cab_count=cali_cab.country.count()
cali_cab_count 

7452

In [6]:
# What is the average rating?
cali_cab.points.mean()

88.47839506172839

In [7]:
# What is the average price?
cali_cab.price.mean()

52.38737040527804

In [8]:
tim="timothy"

 There are  {{tim}} California Cabs in our data set.

### There are a total of 7452 Californa Cab in our data set, with an average rating of 88.49 and and Average Price of $52.39

In [9]:
# For every California Cab in the Data set, 
# Returns the Rating,
# returns the average price based on Rating, 
# along with the total number of wines with that rating

cali_cab_above_90_avg = []
for x in range (90,101):
    cali_cab_above_90 =cali_cab[cali_cab.points == x]
    cali_cab_above_90_avg.append((x,cali_cab_above_90.price.mean(),cali_cab_above_90.price.count()))
    
cali_cab_above_90_avg   

[(90, 53.57654320987654, 810),
 (91, 58.878406708595385, 477),
 (92, 66.83512064343164, 746),
 (93, 79.36652542372882, 472),
 (94, 91.12602739726027, 365),
 (95, 105.59685863874346, 191),
 (96, 113.68354430379746, 79),
 (97, 163.73809523809524, 42),
 (98, 111.66666666666667, 6),
 (99, 108.33333333333333, 3),
 (100, 205.0, 3)]

In [10]:
cali_cab

Unnamed: 0 country                                        description  \
0                0      US  This tremendous 100% varietal wine hails from ...   
16              16      US  This blockbuster, powerhouse of a wine suggest...   
47              47      US  Blended with 9% Malbec, 9% Cabernet Franc and ...   
67              67      US  From the producer's monumental Atlas Peak vine...   
69              69      US  A juiciness of cherry and vanilla spark the op...   
70              70      US  Sweetened tannins highlight a depth of chocola...   
78              78      US  This wine over-delivers in quality for the pri...   
87              87      US  From a multiplicity of vineyards, this wine wo...   
90              90      US  Bright, light oak shadings dress up this mediu...   
97              97      US  Blended with small amounts of Merlot, Cabernet...   
101            101      US  Brambly and herbal, this wine shows bright aci...   
112            112      US  This mountain-grown wine is hugely tannic and ...   
127            127      US  This is a thick, chocolatey wine with a side o...   
134            134      US  While Cabernet Sauvignon dominant, all five re...   
178            178      US  Juicy and soft in blackberry and black cherry,...   
231            231      US  From a seven-acre section of the vineyard, thi...   
284            284      US  Made from 100% Cabernet, this wine shows the i...   
313            313      US  This impressive 100% varietal wine shows focus...   
315            315      US  From a prime site relatively high in elevation...   
324            324      US  This single-vineyard estate wine is made by Ni...   
332            332      US  Dense, rich red fruit belies an elegant wine t...   
341            341      US  Bittersweet at first, with sizable tannins, th...   
379            379      US  A slightly herbal, minty character gives perso...   
380            380      US  An ambitious, concentrated, but not oaky wine,...   
386            386      US  From several vineyard sites and combining all ...   
387            387      US  Grippy leather and tobacco provide a gravelly ...   
389            389      US  Seductive, spicy oak in the aromas and flavors...   
391            391      US  Juicy blackberry and blackcurrant mix with coc...   
406            406      US  Malbec, Merlot and Petit Verdot are included i...   
414            414      US  This straightforward wine shows an exuberance ...   
...            ...     ...                                                ...   
149799      149799      US  You'll recognize the aromas right off the bat:...   
149806      149806      US  Carmel Valley is a little pocket of sunny warm...   
149835      149835      US  This is one of the ripest, least earthy Dry Cr...   
149892      149892      US  I liked the winery's 2001, although it was qui...   
149938      149938      US  From a winery in the heart of Napa that has lo...   
149943      149943      US  Under the warm Lodi sun Cabernet gets as ripe ...   
149950      149950      US  This is Brander's best Cabernet yet. That's ma...   
149964      149964      US  This single-vineyard bottling is riper and mor...   
149974      149974      US  Starts with nice cassis and black-currant arom...   
149982      149982      US  It hits all the right Cabernet notes of blackb...   
150006      150006      US  Clos Pegase's regular Cab, as opposed to the s...   
150012      150012      US  Full-bodied, soft and approachable, with rich ...   
150031      150031      US  Has some real varietal Cabernet notes of curra...   
150038      150038      US  The cherry and smoke aromas wear an attractive...   
150043      150043      US  Rough, earthy and young, with the sharp acidit...   
150046      150046      US  Inexpensive Cabernet can often be slightly wee...   
150054      150054      US  Smells herbaceous and weedy, with lots of gree...   
150113      150113      US  Refinement 